### PR TITLE
Merge ES2K changes

### PR DESCRIPTION
**Pull request by ffoulkes**

### CODE REVIEW

File: switchapi/es2k/switch_pd_p4_name_mapping.h [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+        "linux_networking_control.vxlan_decap_mod_table"
+
+#define LNW_VXLAN_DECAP_MOD_TABLE_KEY_VENDORMETA_MOD_DATA_PTR \
+          "vmeta.common.mod_blob_ptr"
+
+#define LNW_VXLAN_DECAP_MOD_TABLE_ACTION_VXLAN_DECAP_OUTER_IPV4 \
+        "linux_networking_control.vxlan_decap_outer_ipv4"
+
+/* RIF_MOD_TABLE */
+//Verified for MEV - 3 tables instead of 1
```
Comment (nupuruttarwar) [inline switchapi/es2k/switch_pd_p4_name_mapping.h:58]: ES2100

File: switchapi/es2k/switch_pd_p4_name_mapping.h [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+        "vmeta.common.mod_blob_ptr"
+
+#define LNW_NEIGHBOR_MOD_TABLE_ACTION_SET_OUTER_MAC \
+        "linux_networking_control.set_outer_mac"
+#define LNW_ACTION_SET_OUTER_MAC_PARAM_DST_MAC_ADDR \
+        "dst_mac_addr"
+
+
+/* IPV4_TUNNEL_TERM_TABLE */
+//Verified for MEV
```
Comment (nupuruttarwar) [inline switchapi/es2k/switch_pd_p4_name_mapping.h:104]: ES2100?

File: switchapi/es2k/switch_pd_p4_name_mapping.h [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+#define LNW_L2_FWD_TX_TABLE_ACTION_SET_TUNNEL \
+        "linux_networking_control.set_tunnel"
+#define LNW_ACTION_SET_TUNNEL_PARAM_TUNNEL_ID \
+        "tunnel_id"
+#define LNW_ACTION_SET_TUNNEL_PARAM_DST_ADDR \
+        "dst_addr"
+
+
+/* NEXTHOP_TABLE */
+//Verified for MEV
```
Comment (nupuruttarwar) [inline switchapi/es2k/switch_pd_p4_name_mapping.h:164]: ES2100?

File: switchapi/es2k/switch_pd_p4_name_mapping.h [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+
+/* Only 3 bits is allocated for hash size per group in LNW.p4
+ * check LNW_ECMP_HASH_TABLE_KEY_META_COMMON_HASH */
+#define LNW_ECMP_PER_GROUP_HASH_SIZE 8
+
+/* IPV4_TABLE */
+#define LNW_IPV4_TABLE \
+        "linux_networking_control.ipv4_table"
+
+//TODO_LNW_MEV: One additional key for MEV(ipv4_table has 2 keys for MEV)
```
Comment (nupuruttarwar) [inline switchapi/es2k/switch_pd_p4_name_mapping.h:206]: ES2100?

File: switchapi/es2k/switch_pd_routing.c [hunk 5]
```diff
@@ -0,0 +1,1372 @@
...
+            status = tdi_data_field_id_with_action_get(table_info_hdl,
+                                                       LNW_ACTION_SET_NEXTHOP_ID_PARAM_NEXTHOP_ID,
+                                                       action_id, &data_field_id);
+            if (status != TDI_SUCCESS) {
+            krnlmon_log_error("Unable to get data field id param for: %s, error: %d",
+                     LNW_ACTION_SET_NEXTHOP_ID_PARAM_NEXTHOP_ID, status);
+                goto dealloc_resources;
+            }
+
+            // For MEV we need to program nhop_id action in host byte order
```
Comment (nupuruttarwar) [inline switchapi/es2k/switch_pd_routing.c:1072]: ES2100?

File: switchapi/es2k/switch_pd_routing.c [hunk 6]
```diff
@@ -0,0 +1,1372 @@
...
+            status = tdi_key_field_set_value(key_hdl, field_id_meta_bit32_zero,
+                                             0);
+            if (status != TDI_SUCCESS) {
+                krnlmon_log_error("Unable to set value for key ID: %d for ecmp_hash_table"
+                         ", error: %d", field_id_meta_bit32_zero, status);
+                goto dealloc_resources;
+            }
+
+            if (entry_add) {
+                // For MEV we need to program nhop_id action in host byte order
```
Comment (nupuruttarwar) [inline switchapi/es2k/switch_pd_routing.c:1328]: ES2100?

File: switchapi/es2k/switch_pd_utils.c [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+                                              rmac_entry->mac.mac_addr[2],
+                                              rmac_entry->mac.mac_addr[3],
+                                              rmac_entry->mac.mac_addr[4],
+                                              rmac_entry->mac.mac_addr[5]);
+    mac_str[SWITCH_PD_MAC_STR_LENGTH-1] = '\0';
+
+    bf_status = bf_pal_get_port_id_from_mac(bf_dev_id, mac_str, &port_id);
+    if (bf_status != BF_SUCCESS) {
+        // First SWITCH_PD_TARGET_VPORT_OFFSET entries are reserved for
+        // MEV h/w ports. Hence VSI ID/Port ID should be offset with
```
Comment (nupuruttarwar) [inline switchapi/es2k/switch_pd_utils.c:236]: ES2100?

**Comment by ffoulkes** [comment]
> E2100

The day before we are supposed to merge is categorically the wrong time to be making sweeping last-minute changes, especially in symbols and text strings that are part of an interface. The chances of our breaking the code in the process are too high. And ES2K is deeply embedded in the code base -- file names, class names, enum names, target names, etc.

Changes like this should never be made in haste. If we break something in the process, it's on us, the developers, because we should know better. (I _do_ know better, and part of my job is to pipe up when in situations such as this.)

It is _far_ better to leave things as they are for the time being, and make any changes we deem necessary _after_ the code has been open-sourced. It'll give us time to decide what will impacted, to get everybody else on board, and to decide where to draw the line. Do we rename files? Classes? I say no. What's done is done. Do we rename symbols used in interfaces? No, we plan a transition that includes a backward-compatibility plan. We publicize our plan internally, so everyone who's impacted has an opportunity to adapt. And then we make the change publicly.

Because that's the professional thing to do.

**Pull request closed without merging**